File: backend/app/models/alert.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Alert(Base):
    __tablename__ = "alerts"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    nombre = Column(String(255), nullable=False)
    descripcion = Column(Text, nullable=True)
    tipo = Column(String(20), nullable=False)
    estado = Column(String(20), nullable=False, server_default="PENDIENTE")
    fecha_inicio = Column(DateTime, nullable=False)
    fecha_fin = Column(DateTime, nullable=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime, server_default=func.now())

    user = relationship("User", backref="alerts")

File: frontend/src/models/Alert.ts
export interface Alert {
  id: number;
  nombre: string;
  descripcion: string | null;
  tipo: 'REPORTE' | 'ALERTA';
  estado: 'PENDIENTE' | 'FINALIZADO';
  fecha_inicio: string;
  fecha_fin: string | null;
  user_id: number;
  created_at: string;
}

export interface AlertFilters {
  search?: string;
  tipo?: string;
  estado?: string;
}
